demo/DemoArticles/ArticleRef.js

```diff
@@ -0,0 +1,108 @@
+
+/**
+ * Type representing article source endpoints.
+ * @enum {ArticleRefSourceType}
+ */
+const ArticleRefSourceType = {
+  mobileView: 1,
+  mobileContentService: 2
+}
+
+/**
+ * Display name for a given ArticleRefSourceType.
+ * @param {!ArticleRefSourceType} type
+ * @return {!string}
+ */
+const displayNameForArticleRefSourceType = type => {
+  switch (type) {
+  case ArticleRefSourceType.mobileView:
+    return 'MV'
+  case ArticleRefSourceType.mobileContentService:
+    return 'MCS'
+  default:
+    return ''
+  }
+}
+
+/**
+ * Model for article data in ./articles.json
+ */
+class ArticleRef {
+  /**
+   * Constructor
+   * @param {!string} lang
+   * @param {!string} title
+   * @param {!number} revision
+   * @param {!ArticleRefSourceType} sourceType
+   * @return {!object}
+   */
+  constructor(lang, title, revision, sourceType) {
+    this.lang = lang
+    this.title = title
+    this.revision = revision
+    this.sourceType = sourceType
+  }
+
+  /**
+   * String which can be used to for serialization.
+   * @return {!string}
+   */
+  fileName() {
```

Comment:
I believe N should be lowercase but obviously not a show-stopper.

https://en.wikipedia.org/wiki/Filename